internal/host/validator.go

```diff
@@ -372,6 +398,27 @@ func (v *validator) diskEncryptionRequirementsSatisfied(c *validationContext) Va
 		return ValidationPending
 	}
 
+	if hostutil.IsDay2Host(c.host) {
+		luks, err := v.getDiskEncryptionForDay2(c.host)
+		if err != nil {
+			return ValidationPending
+		}
+		if luks == nil {
+			// Disk encryption is disabled for workers on day1 cluster
+			return ValidationSuccessSuppressOutput
+		}
+
+		c.cluster.DiskEncryption = &models.DiskEncryption{}
+		if swag.BoolValue(luks.Clevis.Tpm2) {
+			// If Tpm2 is enabled for workers, check whether supported by the host.
+			c.cluster.DiskEncryption.Mode = swag.String(models.DiskEncryptionModeTpmv2)
+			return boolValue(c.inventory.TpmVersion == models.InventoryTpmVersionNr20)
+		}
+
+		c.cluster.DiskEncryption.Mode = swag.String(models.DiskEncryptionModeTang)
```

Comment:
This looks a bit wired, consider having a switch-case for `tpm` and `tang` and return a failing validation if non of the above is the encryption mode.

Reply:
Actually the Clevis object in Luks contains only Tpm2 and Tang: https://github.com/coreos/ignition/blob/0cbe33fee45d012515479a88f0fe94ef58d5102b/config/v3_2/types/schema.go#L5
I'll make the check more explicit then, in case Tang is empty.